text/0005-media-webrtc-sdk.md

```diff
@@ -0,0 +1,589 @@
+
+## 4.4 Ice-tricle
+
+Each time the client's WebRTC connection has a new ice-candidate, it should be sent to the gateway using the following endpoint:
+
+**_Endpoint_**: POST `GATEWAY/webrtc/conns/:conn_id/ice-remote`
+
+**_Body_**: candidate String
+
+**_Response Data_**: None
+
+## 4.5 Datachannel Request/Response format
+
+The format for encoding all requests and responses sent over the data channel is JSON. The structure of the request and response objects is as follows:
+
+Request/Event:
+
+```
+{
+    type: "event" | "request",
+    seq: Number,
+    cmd: String,
+    data: Option<JSON>,
+}
+```
+
+Response:
+
+```
+{
+    type: "answer",
+    seq: Number,
+    success: bool,
+    error_code: Option<String>,
+    error_msg: Option<String>,
+    data: Option<JSON>,
+}
+```
+
+The seq is an incremental value generated on the sender side. It helps us to map between requests and responses and also detect data loss.
+
+The cmd is generate with rule: `identify.action`, for example:
+
+- `peer.updateSdp`.
+- `sender.{sender_id}.toggle`.
+- `receiver.{receiver_id}.switch`.
+- `room.peers.subscribe`.
+- `room.peers.unsubscribe`.
+- `session.disconnect`.
+- `session.features.mix-minus.add_source`.
+
+## 4.6 In-session requests
+
+
+At the current state, we only have one WebRTC connection to the server, so there is no need to send any requests over HTTP. All requests will be sent over the WebRTC datachannel.
+
+Typically, the client will need to perform various actions with the media server, such as:
+
+- Creating/Releasing senders
+- Creating/Releasing receivers
+- Sender actions: pause, resume, switch stream
+- Receiver actions: pause, resume, switch remote source, update priority, and layers
+
+All actions that involve changing tracks will be performed locally first, and then the `updateSdp` command will be sent to the server.
+
+### 4.6.1 UpdateSDP
+
+Each time we make changes to the WebRTC connection or negotiationneeded event fired, we need to send an `updateSdp` request to the server over the data channel. This request is described below:
+
+**_Cmd:_**: `peer.updateSdp`
+
+**_Data:_**
+
+```
+{
+    sdp: String,
+    tracks: {
+        receivers: [
+            {
+                kind: "audio" | "video",
+                id: String,
+            }
+        ],
+        senders: [
+            {
+                kind: "audio" | "video",
+                id: String,
+                uuid: String,
+                label: String,
+                screen: Option<bool>,
+            }
+        ],
+    }
+}
+```
+
+**_Response data_**:
+
+```
+{
+    sdp: String
+}
+```
+
+### 4.6.2 Room actions, event
+
+We can subscribe to peers event (joined, left, track added, track removed) and also can unsubscribe from it.
```

Comment:
Should the peer events be described in the RFC because there isn't any.

Reply:
Each types: sender, receiver, or features will have events list